### PR TITLE
Update CCC to Conditional Compilation doc

### PR DESCRIPTION
**Pull request by PorterNan**
# What

Update CCC to Conditional Compilation doc

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: change-beta/@azure-communication-react-26897390-5615-4ae3-8818-289d5617a094.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "type": "pre-release",
+  "type": "prerelease",
```
Comment (PorterNan) [inline change-beta/@azure-communication-react-26897390-5615-4ae3-8818-289d5617a094.json:2]: This is kinda blocking rush changelog, have to remove '-' to get it unblocked

File: docs/references/beta-only-features.md [hunk 2]
```diff
@@ -37,6 +64,13 @@ rush switch-flavor:stable
 rush build
 ```
 
+To switch back to beta-release flavor:
+```bash
+rush switch-flavor:beta
```
Comment (prabhjot-msft) [inline docs/references/beta-only-features.md:69]: shouldn't this be `rush switch-flavor:beta-release`

**Review by mgamis-msft — APPROVED** [review]

**Review by prabhjot-msft — APPROVED** [review]
Minor comment, otherwise looks good!

**Review by JamesBurnside — APPROVED** [review]

**Comment by JamesBurnside** [comment]
Change looks good, setting this to merge to cleanup old PRs. We can iterate if there's future changes needed.
cc @PorterNan

**Pull request merged**